File: go/gildedrose/gildeditems/conjured_item.go
package gildeditems

type ConjuredItem struct {
	*Item
}

func (item *ConjuredItem) DoItemUpdate() {
	item.decQuality()
	item.decQuality()
	item.SellIn--
	if item.SellIn < 0 {
		item.decQuality()
		item.decQuality()
	}
}

File: Java/src/main/java/com/gildedrose/Items/Concrete/ConjuredItem.java
package com.gildedrose.Items.Concrete;

import com.gildedrose.Item;
import com.gildedrose.Items.GildedItem;

public class ConjuredItem extends GildedItem {
    public ConjuredItem(Item item) {
        super(item);
    }

    @Override
    public void updateItemQuality() {
        decrementItemQuality();
        decrementItemQuality();
        item.sellIn--;
        if (item.sellIn < 0) {
            decrementItemQuality();
            decrementItemQuality();
        }
    }
}